File: 1894-merge-strings-alternately/1894-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        word1Idx = 0
        word2Idx = 0
        ans = ""
        while word1Idx < len(word1) and word2Idx < len(word2):
            ans += word1[word1Idx]
            word1Idx += 1
            ans += word2[word2Idx]
            word2Idx += 1
        if word1Idx <  len(word1):
            ans += word1[word1Idx:]

        if word2Idx < len(word2):
            ans += word2[word2Idx:]
            
        return ans

File: 1894-merge-strings-alternately/1894-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int word1_idx = 0 , word2_idx = 0;
        string ans;
        while (word1_idx < word1.size() && word2_idx < word2.size()) {
            ans.push_back(word1[word1_idx++]);
            ans.push_back(word2[word2_idx++]);
        }
        while (word1_idx < word1.size()) {
            ans.push_back(word1[word1_idx++]);
        }
        while (word2_idx < word2.size()) {
            ans.push_back(word2[word2_idx++]);
        }
        return ans;
    }
};
